pkg/agent/toggles/types.go

```diff
@@ -66,16 +66,20 @@ func New() *Toggles {
 
 // getMap attempts to resolve the named map toggle against the specified Entity.
 func (t *Toggles) getMap(name string, entity *Entity) map[string]string {
-	if toggle, ok := t.Maps[name]; ok {
-		return toggle(entity)
+	if t != nil && t.Maps != nil {
+		if toggle, ok := t.Maps[name]; ok {
+			return toggle(entity)
+		}
 	}
 	return map[string]string{}
 }
 
 // getString attempts to resolve the named string toggle against the specified Entity.
 func (t *Toggles) getString(name string, entity *Entity) string {
-	if toggle, ok := t.Strings[name]; ok {
-		return toggle(entity)
+	if t != nil && t.Strings != nil {
```

Comment:
could t be nil here?

Reply:
in theory yes, since we need to make this an exported type for RP integration, someone could theoretically do:

var tgls *Toggles
m := tgls.getMap("something", EntityFromWhatever(...)) - this would panic without the above checks

from the perspective of usage within the actual agentbaker client implementation, we should be safeguarded against nil's since we only export the constructor and not the underlying type, which means clients would only be able to interact with the type via the AgentBaker interface after calling the corresponding constructor -  and we make sure to properly instantiate an empty set of toggles in said constructor

Reply:
If I were implementing this, I would return an error when t=nil, so that the client knows while ensuring panic safe. But I don't have concern with this implementation either. ✔️

Reply:
fair enough, this is something I thought about as well, though I think it would probably be better for now to just log something out instead of throwing an error since we'd still want to fulfill the request I think